prow/plugins/trigger/push.go

```diff
@@ -59,11 +59,33 @@ func createRefs(pe github.PushEvent) prowapi.Refs {
 	}
 }
 
+func getPullRequests(c githubClient, pe github.PushEvent) ([]github.PullRequest, error) {
+	org := pe.Repo.Owner.Login
+	repo := pe.Repo.Name
+	for _, commit := range pe.Commits {
+		if commit.ID == "" {
+			continue
+		}
+		pulls, err := c.ListCommitPullRequests(org, repo, commit.ID)
```

Comment:
I think we need to be aggregating pulls for all commits vs. returning pulls for the first commit that has any

Reply:
Thank you for the review. Fixed.
My tests seem to show the GH API bug is still there, and the 1 second sleep helps work around it.